tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import pytest
+from ocs_ci.ocs import ocp, constants, defaults
+from ocs_ci.framework.testlib import tier4c, E2ETest
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import (
+    get_all_pods,
+    check_toleration_on_pods,
+    wait_for_pods_to_be_running,
+)
+from ocs_ci.ocs.node import (
+    taint_nodes,
+    untaint_nodes,
+    get_all_nodes,
+)
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.framework.pytest_customization.marks import bugzilla
+
+
+logger = logging.getLogger(__name__)
+
+
+@tier4c
+@bugzilla("1992472")
+@pytest.mark.polarion_id("OCS-2705")
+class TestNonOCSTaintAndTolerations(E2ETest):
+    """
+    Test to test non ocs taints on ocs nodes
+    and toleration
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, nodes, request):
+        """
+        Make sure all nodes are up and untainted
+
+        """
+
+        def finalizer():
+            nodes.restart_nodes_by_stop_and_start_teardown()
+            assert untaint_nodes(taint_label="xyz=true:NoSchedule"), "Failed to untaint"
+
+        request.addfinalizer(finalizer)
+
+    def test_non_ocs_taint_and_tolerations(self, nodes):
+        """
+        1. Check if nodes are not ocs tainted
+        2. Taint ocs nodes with non-ocs taint
+        3. Set tolerations on storagecluster, subscription and configmap
+        4. Add Capacity
+        5. Respin all ocs pods and check if it runs on ocs nodes with tolerations
+        6. Reboot worker nodes
+
+        """
+
+        # Check tolerations on pods under openshift-storage
+        assert not check_toleration_on_pods(), "Nodes already ocs tainted"
```

Comment:
IMO I dont think we need to check toleration on pod because in this particular testcase we set tolerations on subscription, configmap and storagecluster and not on Pods.
Even if we need check_toleration_on_pods we cannot use it in the beginning of the func as it will not have tolerations for our taint and also by default the func checks for ocs tolerations.